File: src/test/java/com/tjh/riskfactor/controller/LoginControllerTest.java
package com.tjh.riskfactor.controller;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

import org.springframework.test.web.servlet.MockMvc;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.beans.factory.annotation.Autowired;
import static org.springframework.http.HttpHeaders.AUTHORIZATION;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import com.tjh.riskfactor.TestBase;
import static com.tjh.riskfactor.util.Utils.fromJson;

class LoginControllerTest extends TestBase {

    @Test
    void testLogin(@Autowired MockMvc mvc) throws Exception {
        // 不存在的用户
        login(mvc, "what", "the-fuck").andExpect(status().isNotFound());
        // 密码错误
        login(mvc, "admin", "a").andExpect(status().isUnauthorized());
        // 成功
        token(mvc, "admin", "admin");
    }

}

File: backend/src/test/kotlin/com/tjh/riskfactor/controller/LoginControllerTest.kt
package com.tjh.riskfactor.controller

import org.junit.jupiter.api.Test

import org.springframework.test.web.servlet.MockMvc
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

import com.tjh.riskfactor.extension.login
import com.tjh.riskfactor.extension.token

@SpringBootTest
@AutoConfigureMockMvc
class LoginControllerTest {

    @Test
    fun testLogin(@Autowired mvc: MockMvc) {
        // 不存在的用户
        mvc.login("what", "the-fuck").andExpect(status().isNotFound)
        // 密码错误
        mvc.login("admin", "a").andExpect(status().isUnauthorized)
        // 成功
        mvc.token("admin", "admin")
    }

}

